lib/oxttl/src/turtle.rs

```diff
@@ -267,6 +268,130 @@ impl TurtleParser {
     }
 }
 
+pub struct ParallelTurtleParser {
+    parser: TurtleParser,
+    base: Option<Iri<String>>,
+}
+
+impl ParallelTurtleParser {
+    /// Builds a new [`ParallelTurtleParser`].
+    pub fn new() -> Self {
+        ParallelTurtleParser {
+            parser: TurtleParser::new(),
+            base: None,
+        }
+    }
+
+    /// Assumes the file is valid to make parsing faster.
+    ///
+    /// It will skip some validations.
+    ///
+    /// Note that if the file is actually not valid, then broken RDF might be emitted by the parser.
+    #[inline]
+    #[must_use]
+    pub fn unchecked(mut self) -> Self {
+        self.parser = self.parser.unchecked();
+        self
+    }
+
+    #[inline]
+    pub fn with_base_iri(mut self, base_iri: impl Into<String>) -> Result<Self, IriParseError> {
+        self.base = Some(Iri::parse(base_iri.into())?);
+        Ok(self)
+    }
+
+    /// Parses a Turtle file in parallel from a slice of bytes.
+    /// Intended to work on large documents.
+    /// Can fail if there are prefixes that are not defined at the top of the document.
+    ///
+    /// Count the number of people:
+    /// ```
+    /// use oxrdf::vocab::rdf;
+    /// use oxrdf::NamedNodeRef;
+    /// use oxttl::ParallelTurtleParser;
+    /// use rayon::iter::{IntoParallelIterator, ParallelIterator};
+    ///
+    /// let file = br#"@base <http://example.com/> .
+    /// @prefix schema: <http://schema.org/> .
+    /// <foo> a schema:Person ;
+    ///     schema:name "Foo" .
+    /// <bar> a schema:Person ;
+    ///     schema:name "Bar" ."#;
+    ///
+    /// let schema_person = NamedNodeRef::new("http://schema.org/Person")?;
+    /// let readers = ParallelTurtleParser::new().parse_slice(file.as_ref())?;
+    /// let count = readers
+    ///     .into_par_iter()
+    ///     .map(|reader| {
+    ///         let mut count = 0;
+    ///         for triple in reader {
+    ///             let triple = triple.unwrap();
+    ///             if triple.predicate == rdf::TYPE && triple.object == schema_person.into() {
+    ///                 count += 1;
+    ///             }
+    ///         }
+    ///         count
+    ///     }).sum();
+    /// assert_eq!(2, count);
+    /// # Result::<_,Box<dyn std::error::Error>>::Ok(())
+    /// ```
+    #[allow(clippy::unwrap_in_result)]
+    pub fn parse_slice(
+        mut self,
+        slice: &[u8],
+    ) -> Result<Vec<FromSliceTurtleReader<'_>>, TurtleParseError> {
+        const MIN_PAR_SIZE: usize = 10_000;
```

Comment:
Instead of a `MIN_PAR_ITER` that is global what about splitting into something like `max(target_parallelism, slice.len() / 16384)`. An other option might be to always try to split by `target_parallelism` and let the user deal with minimal chunk size if they want

Reply:
Should be fixed now.